src/Polly.Core/Timeout/TimeoutRejectedException.cs

```diff
@@ -0,0 +1,80 @@
+#if !NETCOREAPP
+using System.Runtime.Serialization;
+#endif
+
+namespace Polly.Timeout;
+
+/// <summary>
+/// Exception thrown when a delegate executed through a timeout resilience strategy does not complete, before the configured timeout.
+/// </summary>
+#if NETSTANDARD2_0
+[Serializable]
+#endif
+public class TimeoutRejectedException : ExecutionRejectedException
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TimeoutRejectedException" /> class.
+    /// </summary>
+    public TimeoutRejectedException()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TimeoutRejectedException"/> class.
+    /// </summary>
+    /// <param name="message">The message.</param>
+    public TimeoutRejectedException(string message)
+        : base(message)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TimeoutRejectedException"/> class.
+    /// </summary>
+    /// <param name="message">The message.</param>
+    /// <param name="innerException">The inner exception.</param>
+    public TimeoutRejectedException(string message, Exception innerException)
+        : base(message, innerException)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TimeoutRejectedException" /> class.
+    /// </summary>
+    /// <param name="timeout">The timeout value that caused this exception.</param>
+    public TimeoutRejectedException(TimeSpan timeout) => Timeout = timeout;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TimeoutRejectedException"/> class.
+    /// </summary>
+    /// <param name="timeout">The timeout value that caused this exception.</param>
+    /// <param name="message">The message.</param>
+    public TimeoutRejectedException(string message, TimeSpan timeout)
+        : base(message) => Timeout = timeout;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TimeoutRejectedException"/> class.
+    /// </summary>
+    /// <param name="message">The message.</param>
+    /// <param name="timeout">The timeout value that caused this exception.</param>
+    /// <param name="innerException">The inner exception.</param>
+    public TimeoutRejectedException(string message, TimeSpan timeout, Exception innerException)
+        : base(message, innerException) => Timeout = timeout;
+
+#if !NETCOREAPP
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TimeoutRejectedException"/> class.
+    /// </summary>
+    /// <param name="info">The information.</param>
+    /// <param name="context">The context.</param>
+    protected TimeoutRejectedException(SerializationInfo info, StreamingContext context)
+        : base(info, context)
+    {
+    }
+#endif
+
+    /// <summary>
+    /// Gets the timeout value that caused this exception.
+    /// </summary>
+    public TimeSpan Timeout { get; private set; } = TimeoutConstants.NoTimeout;
```

Comment:
For `NETSTANDARD2_0` we need to read the value of `Timeout` out of the serialization info to restore it in the constructor and write it out in `GetObjectData()`.

Reply:
True. But this is only partial support.

Either we remove the support entirely, like in the other PR, and suppress the warnings, or we should implement it fully.

I would err on the side of removing it.

Reply:
I have fixed this to serialize delay correctly. We can drop the serialization support in separate PR.